app/helpers/ontology_version_helper.rb

```diff
@@ -0,0 +1,32 @@
+module OntologyVersionHelper
+  def failed?(ontology_version)
+    ontology_version.state == 'failed'
+  end
+
+  def latest_version?(ontology_version)
+    ontology_version.ontology.versions.last == ontology_version
+  end
+
+  def can_edit?(ontology, ontology_version)
+    can?(:write, ontology.repository) &&
```

Comment:
You don't need the argument `ontology` here. There is an association `repository` on `OntologyVersion`.